src/mixings/abstract_mixing.h

```diff
@@ -111,6 +108,7 @@ class AbstractMixing {
   virtual void set_state_from_proto(
```

Comment:
also here this could be moved to the `BaseMixing`

Reply:
This function is used in the algorithms, so it is not possible.

Reply:
I don't know why it appeared on this line, but I was referring to `get_state_proto`